lib/Lower/Bridge.cpp

```diff
@@ -574,41 +380,328 @@ class FirConverter : public Fortran::lower::AbstractConverter {
+
+    // Unstructured loop header code - generate loop condition.
+    startBlock(info.headerBlock);
+    mlir::Value tripVariable =
+        builder->create<fir::LoadOp>(location, info.tripVariable);
+    mlir::Value zero = builder->create<mlir::ConstantOp>(
+        location, builder->getIntegerAttr(info.loopVariableType, 0));
+    mlir::Value cond = builder->create<mlir::CmpIOp>(
+        location, mlir::CmpIPredicate::sgt, tripVariable, zero);
+    genFIRConditionalBranch(cond, info.bodyBlock, info.successorBlock);
+  }
+
+  /// Generate FIR to end a structured or unstructured increment loop.
+  void genFIRIncrementLoopEnd(IncrementLoopInfo &info) {
+    mlir::Location location = toLocation();
+    if (info.isStructured()) {
+      // End fir.loop.
+      localSymbols.popShadowSymbol();
+      builder->restoreInsertionPoint(info.insertionPoint);
+      assert(info.doLoop.hasLastValue() && "loop must have a result");
+      mlir::Value result = builder->create<fir::ConvertOp>(
+          location, info.loopVariableType, info.doLoop.lastVal()[0]);
+      builder->create<fir::StoreOp>(location, result, info.loopVariable);
+      return;
+    }
+
+    // Unstructured loop - increment loopVariable.
+    mlir::Value loopVariable =
+        builder->create<fir::LoadOp>(location, info.loopVariable);
+    loopVariable =
+        builder->create<mlir::AddIOp>(location, loopVariable, info.stepValue);
+    builder->create<fir::StoreOp>(location, loopVariable, info.loopVariable);
+
+    // Unstructured loop - decrement tripVariable.
+    mlir::Value tripVariable =
+        builder->create<fir::LoadOp>(location, info.tripVariable);
+    mlir::Value one = builder->create<mlir::ConstantOp>(
+        location, builder->getIntegerAttr(info.loopVariableType, 1));
+    tripVariable = builder->create<mlir::SubIOp>(location, tripVariable, one);
+    builder->create<fir::StoreOp>(location, tripVariable, info.tripVariable);
+    genFIRUnconditionalBranch(info.headerBlock);
+  }
+
+  /// Generate structured or unstructured FIR for an IF construct.
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::IfConstruct &) {
+    if (eval.lowerAsStructured()) {
+      // Structured fir.where nest.
+      fir::WhereOp where;
+      mlir::OpBuilder::InsertPoint insertionPoint =
+          builder->saveInsertionPoint();
+      for (auto &e : *eval.evaluationList) {
+        if (auto *s = e.getIf<Fortran::parser::IfThenStmt>()) {
+          // fir.where op
+          genWhereCondition(where, s);
+        } else if (auto *s = e.getIf<Fortran::parser::ElseIfStmt>()) {
+          // otherwise block, then nested fir.where
+          switchInsertionPointToOtherwise(where);
+          genWhereCondition(where, s);
+        } else if (e.isA<Fortran::parser::ElseStmt>()) {
+          // otherwise block
+          switchInsertionPointToOtherwise(where);
+        } else {
+          genFIR(e, false);
```

Comment:
LLVM usually adds a comment `/*blahBlah=*/false` so one can tell what the `false` means without hunting down the declaration.

Reply:
Alright.